.github/workflows/build-check-install.yaml

```diff
@@ -176,6 +194,16 @@ jobs:
               writeLines(readLines(check_log))
               stop("⚠ R CMD check has warnings")
             }
+            if ("${{ inputs.check-note-blocklist }}" == "true") {
+              print("checking for notes")
+              regexes <- ${{ inputs.note-blocklist }}
+              lines = paste(readLines(check_log), collapse = " ")
+              for (r in regexes) {
+                if (grepl(r, lines)) {
+                  stop(paste("Found a NOTE on the blocklist = ", r))
```

Comment:
This will catch and report the first NOTE that was found on the blocklist. We should make this smarter to report _all_ NOTEs that were found in the log.
Also, take a look at `lapply()` in R. It's faster than `for`

Reply:
I took a look, but this function behaves in a strange way:
```R
x <- lapply(regexes, function(r){
    if(grepl(r, lines)){
       print(r)
    } })
```
Just outputs:
```
Warning message:
In grepl(r, lines) :
  argument 'pattern' has length > 1 and only the first element will be used
```

On the other hand
```R
x <- lapply(regexes, function(r){
    nchar(r) })
```
outputs a list of length = 1 that contains `84 95 89 88 61` (which corresponds to lengths of regexes).
I'll continue checking on Monday.

Reply:
I found a solution - it's ready to be reviewed again.